### PR TITLE
fix(windows-agent): Truncate log files

### PR DESCRIPTION
**Pull request by ashuntu**
This PR renames any existing log to `log.old` when the program starts, overwriting the file if it already exists. It also tests that if a log exists a log.old is successfully created.

---

UDENG-2355

### CODE REVIEW

**Comment by ashuntu** [comment]
Seems like the test is failing only on a Linux host, it passes on my Windows machine. It looks like if I run it on Linux it treats the log directory as the current directory and incorrectly creates logs there. I'm not sure what the best approach is to fix this.

File: windows-agent/cmd/ubuntu-pro-agent/main_test.go [hunk 1]
```diff
@@ -42,16 +44,26 @@ func (a *myApp) PublicDir() (string, error) {
...
 
-		wantReturnCode int
+		wantReturnCode        int
+		wantOldLogFileContent *string
 	}{
 		"Run and exit successfully":                                {},
 		"Run and exit successfully despite logs not being written": {logDirError: true},
 
+		// Log file handling
+		"Existing log file has been renamed to old": {existingLogContent: "foo", wantOldLogFileContent: &fooContent},
```
Comment (didrocks) [inline windows-agent/cmd/ubuntu-pro-agent/main_test.go:63]: I would still keep the test case with the empty file, to document explicitely this behaviour on the edge case now that the code has been modified (the fact that we rename it, and the old log file content is empty then).

**Review by didrocks — COMMENTED** [review]
I know that you kept it as draft (and that the CI is failing suspiciously on:

```
--- FAIL: TestRun (0.00s)
    --- FAIL: TestRun/Run_and_exit_successfully_despite_logs_not_being_written (0.10s)
        main_test.go:123: 
            	Error Trace:	/home/runner/work/ubuntu-pro-for-wsl/ubuntu-pro-for-wsl/windows-agent/cmd/ubuntu-pro-agent/main_test.go:123
            	Error:      	file "log.old" exists
            	Test:       	TestRun/Run_and_exit_successfully_despite_logs_not_being_written
            	Messages:   	Old log file should not exist
```

If we want us to have a look together at the start of your day, do not hesitate!

I have an additional comment on the test itself, but code-wise, this looks good :)

**Review by didrocks — APPROVED** [review]
Excellent work!
